### PR TITLE
Add protected route logic

### PR DESCRIPTION
**Pull request by amar-hunar**
This PR adds protected route to the project

## Features:
- add protected route logic

## Technical:
- add `<ProtectedRoute />`
- add `<AppContainer />`
- add `useGetLoggedInPersonnel` hook
- make the type of success data generic in `useGetReactQuery` hook
- add `interviewScheduling.interface.ts`
- add `JQConstants.ts`

### CODE REVIEW

File: src/components/jobQuery/JQConstants.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import type { JQCommSettingsProps } from 'interfaces';
+
+export const InterviewSchedulingSettingsInitState: JQCommSettingsProps = {
+    general: {
+        isSchedulingForQualifiedEnabled: true,
```
Comment (misss-popcorn) [inline src/components/jobQuery/JQConstants.ts:5]: remove it pls

File: src/interfaces/index.ts [hunk 2]
```diff
@@ -18,6 +18,8 @@ export {
     type QuestionOptionProps
 } from './jobQuery.interface';
 
+export { type JQCommSettingsProps } from './interviewScheduling.interface';
+
```
Comment (misss-popcorn) [inline src/interfaces/index.ts:22]: remove this pls

File: src/interfaces/interviewScheduling.interface.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+export interface JQCommSettingsProps {
+    general: {
+        isSchedulingForQualifiedEnabled: boolean;
+        isReschedulingByCandidateEnabled: boolean;
+        isContactingByCandidateEnabled: boolean;
+        isFeedbackOnRejectionEnabled: boolean;
+    };
+    preInterview: {
```
Comment (misss-popcorn) [inline src/interfaces/interviewScheduling.interface.ts:13]: remove this

File: src/containers/AppContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    const { data: personnel, isLoading } = useGetLoggedInPersonnel({
+        enabled: !!token
+    });
+
+    return token && isLoading ? (
+        <></>
+    ) : (
+        <SettingsContext.Provider
+            value={{
+                ...settingsInitialState,
```
Comment (misss-popcorn) [inline src/containers/AppContainer.tsx:20]: why are we copying code from dashboard? we don't need all of this info here. Pls revert

File: src/contexts/SettingsContext.ts [hunk 5]
```diff
@@ -83,8 +82,7 @@ export const settingsInitialState: SettingsContextType = {
                 },
                 vendor: {
```
Comment (misss-popcorn) [inline src/contexts/SettingsContext.ts:83]: pls remove vendor

File: src/containers/AppContainer.tsx (hunk 4 above)
Comment (amar-hunar) [reply src/containers/AppContainer.tsx]: removed.

File: src/contexts/SettingsContext.ts (hunk 5 above)
Comment (amar-hunar) [reply src/contexts/SettingsContext.ts]: removed

**Pull request merged**